### PR TITLE
Add annotation support to pushgateway service

### PR DESCRIPTION
**Pull request by jmccarty3**
Signed-off-by: Justin McCarty <justin.mccarty@syapse.com>

Re-Opening this as the other was closed (https://github.com/helm/charts/pull/15982) and I do not have permissions to re-open closed PRs

Adds support for service annotations to the prometheus push gateway service. Annotations can be utilized to expose the push gateway outside of the cluster with an internal ELB.

### CODE REVIEW

**Review by sstarcher — APPROVED** [review]

**Comment by jmccarty3** [comment]
@cstaud @gianrubio Can I get a review on this please?

File: stable/prometheus-pushgateway/templates/service.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ apiVersion: v1
 kind: Service
 metadata:
   name: {{ template "prometheus-pushgateway.fullname" . }}
+  annotations:
+{{ .Values.service.annotations | toYaml | indent 4 }}
```
Comment (Aisuko) [inline stable/prometheus-pushgateway/templates/service.yaml]: Please get a confirmation on `.Values.service.annotations`, in my opinion, it's should be the same with `serviceAnnotations`

**Review by Aisuko — CHANGES_REQUESTED** [review]
Please kindly check the issue below.

File: stable/prometheus-pushgateway/templates/service.yaml (hunk 1 above)
Comment (cstaud) [reply stable/prometheus-pushgateway/templates/service.yaml]: Yeah this seems to be incorrect.

**Review by cstaud — CHANGES_REQUESTED** [review]

File: stable/prometheus-pushgateway/templates/service.yaml (hunk 1 above)
Comment (Aisuko) [reply stable/prometheus-pushgateway/templates/service.yaml]: Well, Let's make this clear. Should we add the `serviceAnnotations` to the service, right?

Comment (jmccarty3) [reply stable/prometheus-pushgateway/templates/service.yaml]: Thanks for catching that. I have updated the PR.

**Review by Aisuko — APPROVED** [review]
/approve

**Comment by Aisuko** [comment]
Thanks, please help to merge the PR. @cstaud

**Comment by cstaud** [comment]
Thank you! @jmccarty3 :+1:

**Pull request merged**